### PR TITLE
Updates to annotation dictionary

### PR DESCRIPTION
**Pull request by bnatrajan**
Updates to conform to RDE 0.9.0 draft 4 version of the spec:
- Source option "annotation" will now use new annotation dictionary format
- Old format still accessible via "annotation_old"
- Payload annotation properties are now encoded as a flat list of entries from  json schema
- Highest version can now be specific for the payload annotation schema file
- Encode/Decoder changes to support new annotation dictionary

### CODE REVIEW

File: rde_schema_dictionary_gen.py [hunk 1]
```diff
@@ -389,15 +428,16 @@ def get_latest_version(entity):
         return 'v0_0_0'
 
 
-def get_latest_version_as_ver32(entity):
+def to_ver32(version):
     """
-    Returns the latest version of the entity as a PLDM ver32 array of bytes
+    Returns the a version in Redfish forma (e.g. v1_0_0) to a PLDM ver32 uint32
```
Comment (adkarira) [inline rde_schema_dictionary_gen.py]: format

File: rde_schema_dictionary_gen.py [hunk 2]
```diff
@@ -1069,15 +1241,14 @@ def generate_schema_dictionary(source_type, csdl_schema_dirs, json_schema_dirs,
...
-    oemEntityType = ''
+    oem_sources = []
+    oem_entity_type = ''
     entity_repo = {}
     entity_offset_map = {}
     includeNamespaces = {}
 
     # Validate source type.
-    if source_type not in ['local', 'annotation', 'remote']:
+    if source_type not in ['local', 'annotation', 'annotationV2', 'remote']:
```
Comment (adkarira) [inline rde_schema_dictionary_gen.py]: annotationV2 should be annotation_old, right?

File: rde_schema_dictionary_gen.py (hunk 1 above)
Comment (bnatrajan) [reply rde_schema_dictionary_gen.py]: fixed

File: rde_schema_dictionary_gen.py (hunk 2 above)
Comment (bnatrajan) [reply rde_schema_dictionary_gen.py]: fixed

**Pull request merged**